packages/@yoda/bluetooth/a2dp.js

```diff
@@ -0,0 +1,537 @@
+'use strict'
+
+var logger = require('logger')('bluetooth-a2dp')
+var inherits = require('util').inherits
+var EventEmitter = require('events').EventEmitter
+var floraFactory = require('@yoda/flora')
+var FloraComp = require('@yoda/flora/comp')
+var protocol = require('./protocol.json')
+var AudioManager = require('@yoda/audio').AudioManager
+var _ = require('@yoda/util')._
+var stateFilters = require('./a2dp_statemap').getStateFilters()
+
+/**
+ * @typedef {Object} PROFILE
+ * @property {string} BLE - Bluetooth low energy profile.
+ * @property {string} A2DP - Bluetooth advanced audio distribution profile.
+ */
+
+/**
+ * @typedef {Object} A2DP_MODE
+ * @property {string} SINK - A2dp sink mode.
+ * @property {string} SOURCE - A2dp source mode.
+ */
+
+/**
+ * @typedef {Object} RADIO_STATE
+ * @property {string} ON - Bluetooth radio state is ON.
+ * @property {string} OFF - Bluetooth radio state is OFF.
+ * @property {string} ON_FAILED - Turn on bluetooth radio failed.
+ */
+
+/**
+ * @typedef {Object} CONNECTION_STATE
+ * @property {string} CONNECTED - Successfully connected to remote device.
+ * @property {string} DISCONNECTED - Disconnected from remote deivce.
+ * @property {string} CONNECT_FAILED - Failed to connect to remote device.
+ * @property {string} AUTOCONNECT_FAILED - Auto connection to history paired device failed after turn on bluetooth.
+ */
+
+/**
+ * @typedef {Object} AUDIO_STATE
+ * @property {string} PLAYING - Music stream is playing.
+ * @property {string} PAUSED - Music stream is paused.
+ * @property {string} STOPPED - Music stream is stopped.
+ * @property {string} VOLUMN_CHANGED - Music stream's volumn is changed.
+ */
+
+/**
+ * @typedef {Object} DISCOVERY_STATE
+ * @property {string} ON - Local device is discoverable.
+ * @property {string} OFF - Local device is undiscoverable.
+ * @property {string} DEVICE_LIST_CHANGED - Found new around bluetooth devices.
+ */
+
+/**
+ * When bluetooth's radio state is changed, such as on/off.
+ * @event module:@yoda/bluetooth.BluetoothA2dp#radio_state_changed
+ */
+
+/**
+ * When bluetooth's connection state is changed, such as connected/disconnected.
+ * @event module:@yoda/bluetooth.BluetoothA2dp#connection_state_changed
+ */
+
+/**
+ * When bluetooth's audio stream state is changed, such as playing/paused.
+ * @event module:@yoda/bluetooth.BluetoothA2dp#audio_state_changed
+ */
+
+/**
+ * When bluetooth's discovery state is changed, such as undiscoverable or device list changed.
+ * @event module:@yoda/bluetooth.BluetoothA2dp#discovery_state_changed
+ */
+
+/**
+ * Use `bluetooth.getAdapter(PROFILE.A2DP)` instead of this constructor.
+ * @class
+ * @memberof module:@yoda/bluetooth
+ * @constructor

```

Comment:
Class and constructor acts the same role.

Reply:
Modified.